imap_processing/codice/codice_l1a_lo_species.py

```
@@ -0,0 +1,286 @@
+    """
+    # Get these values from unpacked data. These are used to
+    # lookup in LUT table.
+    table_id = np.unique(unpacked_dataset["table_id"].values)[0]
+    view_id = np.unique(unpacked_dataset["view_id"].values)[0]
+    apid = np.unique(unpacked_dataset["pkt_apid"].values)[0]
+    plan_id = np.unique(unpacked_dataset["plan_id"].values)[0]
+    plan_step = np.unique(unpacked_dataset["plan_step"].values)[0]
+
+    logger.info(
+        f"Processing species with - APID: {apid}, View ID: {view_id}, "
+        f"Table ID: {table_id}, Plan ID: {plan_id}, Plan Step: {plan_step}"
+    )
+
+    # ========== Get LUT Data ===========
+    # Read information from LUT
+    sci_lut_data = read_sci_lut(lut_file, table_id)
+    if sci_lut_data is None:
+        raise ValueError(f"SCI-LUT file does not have data for table ID {table_id}.")
+
+    view_tab_info = get_view_tab_info(sci_lut_data, view_id, apid)
+    view_tab_obj = ViewTabInfo(
+        apid=apid,
+        view_id=view_id,
+        sensor=view_tab_info["sensor"],
+        three_d_collapsed=view_tab_info["3d_collapse"],
+        collapse_table=view_tab_info["collapse_table"],
+    )
+
+    if view_tab_obj.sensor != 0:
+        raise ValueError("Unsupported sensor ID for Lo species processing.")
+
+    # ========= Decompress and Reshape Data ===========
+    # Lookup SW or NSW species based on APID
+    if view_tab_obj.apid == CODICEAPID.COD_LO_SW_SPECIES_COUNTS:
+        species_names = sci_lut_data["data_product_lo_tab"]["0"]["species"]["sw"][
+            "species_names"
+        ]
+        logical_source_id = "imap_codice_l1a_lo-sw-species"
+    elif view_tab_obj.apid == CODICEAPID.COD_LO_NSW_SPECIES_COUNTS:
+        species_names = sci_lut_data["data_product_lo_tab"]["0"]["species"]["nsw"][
+            "species_names"
+        ]
+        logical_source_id = "imap_codice_l1a_lo-nsw-species"
+    else:
+        logger.error("Unknown apid in Lo species processing.")
```

raise here instead of log. I'm guessing this won't succeed after this. Also include what the unknown apid is in the message.
```suggestion
        raise ValueError(f"Unknown apid {view_tab_obj.apid} in Lo species processing.")
```